editor/src/messages/portfolio/document/navigation/navigation_message_handler.rs

```diff
@@ -151,7 +159,7 @@ impl MessageHandler<NavigationMessage, (&Document, Option<[DVec2; 2]>, &InputPre
 					responses.add(SetCanvasRotation { angle_radians: self.tilt + rotation });
 				}
 
-				if self.zooming {
+				if self.zooming && !self.panning && !self.tilting {
```

Comment:
Perhaps this should be an enum:
```rs
pub enum TransformOperation {
    None,
    Pan,
    Rotate,
    Zoom
}
```

Reply:
agreed, I found it like this and stuck to chesterton's fence.

Reply:
change made. I moved all the transform operation state into the enum, this way invalid states are harder to represent.